component-lib/src/business/Table/Table.tsx

```diff
@@ -16,30 +14,48 @@ export const TableBodyCell: React.FC<TableBodyCellProps> = (props) => {
     const { children } = props;
 
     return (
-        <td onClick={props.onClick} className={cs({
-            "data-table__cell": true,
-            "data-table__cell--right-aligned": props.rightAligned
-        })}>{children}</td>
+        <td onClick={props.onClick}
+            role={props.onClick ? "button" : undefined}
+            onKeyDown={props.onClick ? (e) => {
```

Comment:
We need to stop propagation here to prevent this event from bubbling and cancel the native checkbox event (or other potential onClicks within the cell)  

```suggestion
onKeyDown={e => {
        e.stopPropagation();
        if (props.onClick && (e.keyCode === 32 || e.keyCode === 13)) {
          e.preventDefault();
          props.onClick && props.onClick();
        }
      }}
```